packages/app-utils/src/components/ConnectionBootstrap.tsx

```diff
@@ -59,39 +58,53 @@ export function ConnectionBootstrap({
     [api, client]
   );
 
-  useEffect(() => {
-    if (connection == null) return;
+  useEffect(
+    function listenForShutdown() {
+      if (connection == null) return;
 
-    // handles the shutdown event
-    function handleShutdown(event: CustomEvent): void {
-      const { detail } = event;
-      log.info('Shutdown', `${JSON.stringify(detail)}`);
-      setError(`Server shutdown: ${detail ?? 'Unknown reason'}`);
-    }
-    const removerFn = connection.addEventListener(
-      api.IdeConnection.EVENT_SHUTDOWN,
-      handleShutdown
-    );
+      // handles the shutdown event
+      function handleShutdown(event: CustomEvent): void {
+        const { detail } = event;
+        log.info('Shutdown', `${JSON.stringify(detail)}`);
+        setError(`Server shutdown: ${detail ?? 'Unknown reason'}`);
+      }
+      const removerFn = connection.addEventListener(
+        api.IdeConnection.EVENT_SHUTDOWN,
+        handleShutdown
+      );
+
+      return removerFn;
+    },
+    [api, connection]
+  );
 
-    // handles the auth failed event
-    function handleAuthFailed(event: CustomEvent): void {
-      const { detail } = event;
-      log.warn('Reconnect authentication failed', `${JSON.stringify(detail)}`);
-      setError(
-        `Reconnect authentication failed: ${detail ?? 'Unknown reason'}`
+  useEffect(
+    function listenForAuthFailed() {
+      if (connection == null) return;
+      // handles the auth failed event
+      function handleAuthFailed(event: CustomEvent): void {
+        const { detail } = event;
+        log.warn(
+          'Reconnect authentication failed',
+          `${JSON.stringify(detail)}`
+        );
+        setError(
+          `Reconnect authentication failed: ${detail ?? 'Unknown reason'}`
+        );
+        setIsAuthFailedState(true);
+      }
+      const authFailed = connection.addEventListener(
```

Comment:
Change to a more descriptive name. `addEventListener` in our API returns a function to remove the event listener

```suggestion
      const removerFn = connection.addEventListener(
```